volatility/framework/plugins/windows/dlllist.py

```diff
@@ -68,19 +71,21 @@ def dump_pe(cls,
             if layer_name is None:
                 layer_name = dll_entry.vol.layer_name
 
-            filedata = interfaces.plugins.FileInterface("{0}.{1:#x}.{2:#x}.dmp".format(
-                ntpath.basename(name), dll_entry.vol.offset, dll_entry.DllBase))
+            file_handle = file_handler("{}{}.{:#x}.{:#x}.dmp".format(prefix, ntpath.basename(name),
+                                                                     dll_entry.vol.offset, dll_entry.DllBase))
 
             dos_header = context.object(pe_table_name + constants.BANG + "_IMAGE_DOS_HEADER",
                                         offset = dll_entry.DllBase,
                                         layer_name = layer_name)
 
-            for offset, data in dos_header.reconstruct():
-                filedata.data.seek(offset)
-                filedata.data.write(data)
-        except Exception as excp:
+            with file_handle as file_data:
```

Comment:
I must be missing something subtle here. Why does this need a with clause? When I don't have one, the file doesn't get renamed.

Reply:
The file handle is open and the renaming only happens when the file is closed (to allow plugins to change the name, for instance if the file handle is returned from a classmethod in another plugin that chooses an inappropriate name).  I'd forgotten this circumstance, but a number of the dumping methods seem like they should be reusable, but to achieve that they have to return an open file handle.  They don't really need the with clause though, I've removed that and changed the self._file_handler argument to self.open, which may be a little unusual in these cirucmstances, since it's both an open_method/contextmanager and also a file constructor but it's somewhat the best option we've got.  5:S